docs/package_templates/autotools_package/all/conanfile.py

```diff
@@ -75,64 +85,99 @@ def requirements(self):
+        if is_msvc(self):
+            self.tool_requires("automake/x.y.z")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def generate(self):
-        # autotools usually uses 'yes' and 'no' to enable/disable options
-        yes_no = lambda v: "yes" if v else "no"
+        # inject tools_requires env vars in build scope (not needed if there is no tool_requires)
+        env = VirtualBuildEnv(self)
+        env.generate()
+        # inject requires env vars in build scope
+        # it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
         # --fpic is automatically managed when 'fPIC'option is declared
         # --enable/disable-shared is automatically managed when 'shared' option is declared
         tc = AutotoolsToolchain(self)
-        tc.configure_args.append("--with-foobar=%s" % yes_no(self.options.with_foobar))
-        tc.configure_args.append("--enable-tools=no")
-        tc.configure_args.append("--enable-manpages=no")
+        # autotools usually uses 'yes' and 'no' to enable/disable options
+        yes_no = lambda v: "yes" if v else "no"
+        tc.configure_args.extend([
+            f"--with-foobar={yes_no(self.options.with_foobar)}",
+            "--enable-tools=no",
+            "--enable-manpages=no",
+        ])
         tc.generate()
-        # generate dependencies for pkg-config
+        # generate pkg-config files of dependencies (useless if upstream configure.ac doesn't rely on PKG_CHECK_MODULES macro)
         tc = PkgConfigDeps(self)
         tc.generate()
         # generate dependencies for autotools
         tc = AutotoolsDeps(self)
         tc.generate()
-        # inject tools_requires env vars in build scope (not needed if there is no tool_requires)
-        env = VirtualBuildEnv(self)
-        env.generate()
-        # inject requires env vars in build scope
-        # it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable
-        if not cross_building(self):
-            env = VirtualRunEnv(self)
-            env.generate(scope="build")
+
+        # If Visual Studio is supported
+        if is_msvc(self):
+            env = Environment()
+            # get compile & ar-lib from automake (or eventually lib source code if available)
+            # it's not always required to wrap CC, CXX & AR with these scripts, it depends on how much love was put in
+            # upstream build files
+            compile_wrapper = unix_path(self, self._user_info_build["automake"].compile)
+            ar_wrapper = unix_path(self, self._user_info_build["automake"].ar_lib)
```

Comment:
We'll need to remember to update these soon when `automake` is updated to use the new `conf` variables.